examples/ExactlyOnce/Program.cs

```diff
@@ -14,789 +14,677 @@
-        /// </summary>
-        const string TransactionalIdPrefix_Aggregate = "aggregator-transaction-id";
+            var wordCreatorConfig = configuration.GetSection("WordCreator").Get<WordCreatorConfig>();
+            var wordReverserConfig = configuration.GetSection("WordReverser").Get<WordReverserConfig>();
+            var adminClientConfig = configuration.GetSection("AdminClient").Get<AdminClientConfig>();
 
+            wordCreatorConfig.Producer.ThrowIfContainsNonUserConfigurable();
+            wordReverserConfig.Producer.ThrowIfContainsNonUserConfigurable();
+            wordReverserConfig.Consumer.ThrowIfContainsNonUserConfigurable();
 
-        /// <summary>
-        ///     Name of the consumer group used in the map phase when consuming
-        ///     from the Topic_InputLines topic.
-        /// </summary>
-        const string ConsumerGroup_MapWords = "map-words-consumer-group";
+            var adminClient = new AdminClientBuilder(adminClientConfig).Build();
+            await CreateTopic(adminClient, wordCreatorConfig.OutputTopic, wordCreatorConfig.TopicPartitions);
+            await CreateTopic(adminClient, wordReverserConfig.OutputTopic, wordReverserConfig.TopicPartitions);
+
+            switch (command)
+            {
+                case "create_words":
+                    wordCreatorConfig.Producer.TransactionalId = transactionalId;
+                    new WordCreator
+                    {
+                        Config = wordCreatorConfig,
+                        CancellationTokenSource = cts
+                    }
+                    .Run();
+                    break;
+                case "reverse_words":
+                    wordReverserConfig.Producer.TransactionalId = transactionalId;
+                    new WordReverser
+                    {
+                        Config = wordReverserConfig,
+                        CancellationTokenSource = cts
+                    }
+                    .Run();
+                    break;
+                default:
+                    logger.LogError("Unknown command");
+                    break;
+            }
+            completedCancellationTokenSource.Cancel();
+        }
+    }
+
+    /// <summary>
+    ///     This transactional producer allows to use a producer an consumer for transactions,
+    ///     by recreating the producer when a fatal error happens, sending offsets to transaction
+    ///     and handling aborts and retries correctly.
+    /// </summary>
+    public class TransactionalProducer<K, V> : IDisposable
+    {
+        public string OutputTopic { get; set; }
+
+        private ILogger logger = Program.LoggerFactory.CreateLogger<TransactionalProducer<K, V>>();
+
+        public TimeSpan RetryInterval { get; set; }
+
+        public TimeSpan LocalTransactionOperationTimeout { get; set; }
+
+        public ProducerConfig ProducerConfig { get; set; }
+
+        public IConsumer<K, V> Consumer { get; set; }
```

Comment:
a thing called "producer" should not have access to a thing called "consumer" - indication of bad structure.